Code/Tools/BundleLauncher/O3DE_SDK_Launcher.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Component/ComponentApplication.h>
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzFramework/Process/ProcessWatcher.h>
+
+#include <cstdlib>
+#include <mach-o/dyld.h>
+
+int main(int argc, char* argv[])
+{
+    // We need to pass in the engine path since we won't be able to find it by searching upwards.
+    // We can't use any containers that use our custom allocator till after the call to ComponentApplication::Create()
+    AZ::IO::FixedMaxPath processPath = AZ::Utils::GetExecutableDirectory();
+    AZ::IO::FixedMaxPath enginePath = (processPath / "../Engine").LexicallyNormal();
+    auto enginePathParam = AZ::SettingsRegistryInterface::FixedValueString::format(R"(--engine-path="%s")", enginePath.c_str());
+    // Uses the fixed_vector deduction guide to determine the type is AZStd::fixed_vector<char*, 3>
+    AZStd::fixed_vector commandLineParams{ processPath.Native().data(), enginePathParam.data(), static_cast<char*>(nullptr) };
```

Comment:
The nullptr argument is not needed.
This isn't invoking `execve` to create another process, so that parameter is superflous.

```suggestion
    // Uses the fixed_vector deduction guide to determine the type is AZStd::fixed_vector<char*, 2>
    AZStd::fixed_vector commandLineParams{ processPath.Native().data(), enginePathParam.data() };
```

Reply:
Fixed.